w06h03/test/pgdp/ds/behavior/GraphTest.java

```diff
@@ -60,31 +50,25 @@ final void directionOfEdgeShouldBeRespected() {
         assertArrayEquals(new int[0], neighbors);
     }
 
-    private int random(int min, int max) {
-        return (int) (Math.random() * (max - min + 1) + min);
+
+    @Test
+    final void shouldNotAddEdgeWhenNodeIsOutOfRange() {
+        final var graph = createGraph(3);
+        graph.addEdge(1, 3);
+        assertFalse(graph.isAdj(1, 3));
     }
 
-    private <T> T switchGraph(T sparse, T dense) {
-        return switch (graphType.getSimpleName()) {
-            case "SparseGraph" -> sparse;
-            case "DenseGraph" -> dense;
-            default -> throw new IllegalArgumentException("Unknown graph type");
-        };
+    @Test
+    final void shouldNotAddEdgeWhenNodeIsNegative() {
```

Comment:
I don't think this behaviour was defined on Artemis. Are you sure we need to check for this?

Reply:
I would rather keep it, just to be sure. Johannes [did not explicitly deny it](https://zulip.in.tum.de/#narrow/stream/1481-PGdP-W06H03/topic/from.20parameter.20g.C3.BCltig.3F).

Reply:
Ready to merge?

Reply:
@nilsreichardt Let's wait for @HendrikHuebner's reply.

Reply:
You can go ahead and merge it, I'll ask on zulip and if they say we dont need to test for it we can still remove it later.